validator/keymanager/remote-web3signer/keymanager.go

```diff
@@ -420,10 +712,21 @@ func DisplayRemotePublicKeys(validatingPubKeys [][48]byte) {
 }
 
 // AddPublicKeys imports a list of public keys into the keymanager for web3signer use. Returns status with message.
-func (km *Keymanager) AddPublicKeys(pubKeys []string) []*keymanager.KeyStatus {
+func (km *Keymanager) AddPublicKeys(pubKeys []string) ([]*keymanager.KeyStatus, error) {
 	importedRemoteKeysStatuses := make([]*keymanager.KeyStatus, len(pubKeys))
+	// Using a map to track both existing and new public keys efficiently
+	combinedKeys := make(map[string][48]byte)
+
+	// Populate the map with existing keys
+	km.lock.RLock()
+	originalKeysLen := len(km.providedPublicKeys)
+	for _, key := range km.providedPublicKeys {
+		encodedKey := hexutil.Encode(key[:])
+		combinedKeys[encodedKey] = key
+	}
+	km.lock.RUnlock()
```

Comment:
We do this a few times, would it be worth extracting to another function to avoid accidentally forgetting a lock?

Reply:
hmm the other time is at the start of the keymanager which shouldn't need the lock I think and the delete which also should have the check for if it's 0. will think about it but open to suggestions on how to pull it out.